### DISCUSSION TITLE
Problems with pipe and custom middleware pattern on route handlers

### DISCUSSION THREAD

**Discussion by bolmos-o** [vercel/next.js#71958] [Help]
### Summary

Our team has been using a `pipe()` function on route handlers to allow for custom middleware to be run. However, this pattern has caused us some problems, e.g. the compiler does not realize it is a dynamic route by its own, throwing errors during build; or the codemod for Next.js 15 simply does not identify the changes to be made. We are currently trying to identify if this pattern we are using is future-proof - although we had these problems -, or if it is going to continuously bring us some unnecessary burden with every Next.js release, which probably means we should drop it. Do you have any insights?

Thanks!

### Additional information

```js
export const GET = pipe(middleware1, middleware2, (req) => {

 ...

 return new Response()
})
```

**Comment by aniruddhaadak80** [comment]
Hi @bolmos-o ,

Using a custom `pipe()` function for middleware in route handlers can lead to issues in Next.js, especially with:

1. **Dynamic Route Recognition**: The compiler may not recognize it as a dynamic route, causing build errors.
2. **Next.js Codemods Compatibility**: Next.js 15 codemods might fail to detect or update these patterns, leading to compatibility issues with future updates.

### Recommendation:
Consider replacing `pipe()` with Next.js’s standard middleware pattern to ensure compatibility, as non-standard patterns may break with future releases.
